testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5334,6 +5334,33 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkScalar("rand_integer(2, 11)", 1, "INTEGER NOT NULL");
   }
 
+  /** Tests {@code ARRAY_APPEND} function from Spark. */
+  @Test void testArrayAppendFunc() {
+    final SqlOperatorFixture f0 = fixture();
+    f0.setFor(SqlLibraryOperators.ARRAY_APPEND);
+    f0.checkFails("^array_append(array[1], 2)^",
+        "No match found for function signature ARRAY_APPEND\\("
+            + "<INTEGER ARRAY>, <NUMERIC>\\)", false);
+
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("array_append(array[1], 2)", "[1, 2]",
+        "INTEGER NOT NULL ARRAY NOT NULL");
+    f.checkScalar("array_append(array[1], null)", "[1, null]",
+        "INTEGER ARRAY NOT NULL");
+    f.checkScalar("array_append(array(null), null)", "[null, null]",
+        "NULL ARRAY NOT NULL");
+    f.checkScalar("array_append(array(), null)", "[null]",
```

Comment:
Can we append integer or string to `array_append(array(), null)`?

(The `UNKNOWN` type maybe trick.）

Reply:
good catch! it should and i add the test array_append(array(), 1) i find a bug , the return type should return leastRestrictive type. i fixed.

Reply:
the result is aligns with spark. some different is return type not null. which is not print in spark.
```
spark-sql (default)> select array_append(array(), 1);
[1]
spark-sql (default)> select typeof(array_append(array(), 1));
array<int>

spark-sql (default)> select array_append(array(), null);
[null]
spark-sql (default)> select typeof(array_append(array(), null));
array<void>

```